### PR TITLE
Compute the icon based on the logbook state and not the current state

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

- If there was no custom icon, we used the current state to
  calculate the icon. We need to use the state from the log

I thought this was a regression caused by #12713 but I 
went back to Monday's build and it was the same.

Before

Screen Shot 2022-05-18 at 15 47 54
Screen Shot 2022-05-18 at 15 41 53

After
Screen Shot 2022-05-18 at 17 37 55
Screen Shot 2022-05-18 at 17 37 45

Screen Shot 2022-05-18 at 16 34 04
Screen Shot 2022-05-18 at 17 09 06

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bdraco** [comment]
At some point fetching the icon started returning svg data ??

**Comment by bdraco** [comment]
So in #10223 we now return an SVG but overrideIcon expects something else

**Comment by bdraco** [comment]
Still need to fix custom icons

File: src/panels/logbook/ha-logbook-renderer.ts [hunk 1]
```diff
@@ -140,26 +140,37 @@ class HaLogbookRenderer extends LitElement {
...
+            entity_picture: DOMAINS_WITH_DYNAMIC_PICTURE.has(domain)
+              ? undefined
+              : currentStateObj?.attributes.entity_picture,
+          },
+        }) : currentStateObj;
+    const overrideImage =
+      !historicStateObj &&
+      !item.icon &&
+      domain &&
+      isComponentLoaded(this.hass, domain)
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/panels/logbook/ha-logbook-renderer.ts:166]: https://github.com/home-assistant/frontend/pull/12713#discussion_r876231915

**Comment by bdraco** [comment]
Need to fix custom images
Screen Shot 2022-05-19 at 12 29 30

**Review by balloob — APPROVED** [review]

**Pull request merged**